biosppy/signals/ecg.py

```diff
@@ -1381,6 +1381,858 @@ def ASI_segmenter(signal=None, sampling_rate=1000.0, Pth=5.0):
+
+    """
+
+    # check inputs
+    if ecg is None:
+        raise TypeError("Please specify an input signal.")
+        
+    ''' Initialize '''
+    delay = 0
+    skip = 0                    # Becomes one when a T wave is detected
+    m_selected_RR = 0
+    mean_RR = 0
+    ser_back = 0
+
+
+    ''' Noise Cancelation (Filtering) (5-18 Hz) '''
+
+    if fs == 200:
+        ''' Remove the mean of Signal '''
+        #If fs=200 keep frwquency 5-12 Hz otherwise 5-18 Hz
+        ecg = ecg - np.mean(ecg)  
+
+        Wn = 12*2/fs
+        N = 3
+        a, b = signal.butter(N, Wn, btype='lowpass')
+        ecg_l = signal.filtfilt(a, b, ecg)
+        
+        ecg_l = ecg_l/np.max(np.abs(ecg_l)) #Normalize by dividing high value. That reduce time of calculation
+
+        Wn = 5*2/fs
+        N = 3                                           # Order of 3 less processing
+        a, b = signal.butter(N, Wn, btype='highpass')             # Bandpass filtering
+        ecg_h = signal.filtfilt(a, b, ecg_l, padlen=3*(max(len(a), len(b))-1))
+        ecg_h = ecg_h/np.max(np.abs(ecg_h))  #Normalize by dividing high value. That reduce time of calculation
+
+    else:
+        ''' Band Pass Filter for noise cancelation of other sampling frequencies (Filtering)'''
+        f1 = 5 #3 #5                                          # cutoff low frequency to get rid of baseline wander
+        f2 = 18 #25  #15                                         # cutoff frequency to discard high frequency noise
+        Wn = [f1*2/fs, f2*2/fs]                         # cutoff based on fs
+        N = 3                                           
+        a, b = signal.butter(N=N, Wn=Wn, btype='bandpass')   # Bandpass filtering
+        ecg_h = signal.filtfilt(a, b, ecg, padlen=3*(max(len(a), len(b)) - 1))
+                    
+        ecg_h = ecg_h/np.max(np.abs(ecg_h))
+
+    vector = [1, 2, 0, -2, -1]
+    if fs != 200:
+        int_c = 160/fs
+        b = interp1d(range(1, 6), [i*fs/8 for i in vector])(np.arange(1, 5.1, int_c))  
+                                                                                    
+    else:
+        b = [i*fs/8 for i in vector]      
+
+    ecg_d = signal.filtfilt(b, 1, ecg_h, padlen=3*(max(len(a), len(b)) - 1))
+
+    ecg_d = ecg_d/np.max(ecg_d)
+
+
+    ''' Squaring nonlinearly enhance the dominant peaks '''
+
+    ecg_s = ecg_d**2
+    
+    #Smooting
+    sm_size = int(0.06 * fs)       
+    ecg_s = smoother(signal=ecg_s, kernel='flattop', size=sm_size, mirror=True)
+
+
+    temp_vector = np.ones((1, round(0.150*fs)))/round(0.150*fs) # 150ms moving window, widest possible QRS width
+    temp_vector = temp_vector.flatten()
+    ecg_m = np.convolve(ecg_s, temp_vector)  #Convolution signal and moving window sample
+
+    delay = delay + round(0.150*fs)/2
+
+
+    pks = []
+    locs = peakutils.indexes(y=ecg_m, thres=0, min_dist=round(0.231*fs))  #Find all the peaks apart from previous peak 231ms, peak indices
```

Comment:
Please confirm if this the PeakUtils package, so it can be added to the optional requirements for Biosppy :)

Reply:
Also don't forget to import the library at the beginning of the file!

Reply:
the peakutils is this one: https://peakutils.readthedocs.io/en/latest/. Sorry for the missing import